electrum/commands.py

```diff
@@ -1475,6 +1475,8 @@ def get_parser():
     # daemon
     parser_daemon = subparsers.add_parser('daemon', help="Run Daemon")
     parser_daemon.add_argument("-d", "--detached", action="store_true", dest="detach", default=False, help="run daemon in detached mode")
+    parser_daemon.add_argument("--unixsock", action="store_true", dest="unixsocket", default=False, help="bind daemon to unix socket")
+    parser_daemon.add_argument("--unixsockpath", dest="unixsocketpath", help="where to place unix socket")
```

Comment:
If we made unix sockets the default on Linux, which to me sounds like a good default, `--unixsock` (noarg) couldn't really be adopted as-is.
We could instead have `--unixsock` that takes a boolean...
Except, sounds like we might (later) add support for Windows named pipes, in which case we might as well do it completely differently now:
how about this:
- `--rpcsock`, which takes one arg, valid arg values for now being `tcp` and `unixsock` (and later also `winpipe`)? and
- `--rpcsockpath`, which also takes one arg, a file system path, used in case of `rpcsock==unixsock` (and later also when `rpcsock==winpipe`)
- the respective config keys can be named identically to the cli/rpc options: `rpcsock` and `rpcsockpath`

Reply:
In Python 3.9, we'll get `argparse.BooleanOptionalAction` in `argparse`:
```
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('--foo', action=argparse.BooleanOptionalAction)
>>> parser.parse_args(['--no-foo'])
Namespace(foo=False)
```

Maybe you could do a poor man's version of that, by making `--unixsock` a no-op and then making `--no-unixsock` the active one, and then just slot in `argparse.BooleanOptionalAction` once Python 3.9 is old enough. It would be a bit ungainly, but only temporary.

As for `--rpcsock`, there is, to my knowledge, not any system where you can have both Windows pipes and Unix sockets. So I don't understand why you would need the triple-barreled option - to me, a boolean (use file/tcp) seems more suitable.

You could however rename the parameter to `--rpcsock` now already, because that way you wouldn't have to deprecate `--unixsock` because it never existed.

Reply:
> As for `--rpcsock`, there is, to my knowledge, not any system where you can have both Windows pipes and Unix sockets. So I don't understand why you would need the triple-barreled option - to me, a boolean (use file/tcp) seems more suitable.

There might already be [unix socket support on Windows](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/), or maybe there will be in the future. Pretty sure they cannot be used for our purposes as of right now though. Nevertheless this was not really my motivation.
Simply, the alternative where the user runs with `--unixsock` on Windows, and it uses something similar (named pipes) but not quite the same, seemed a bit weird.

> making `--unixsock` a no-op and then making `--no-unixsock` the active one, and then just slot in `argparse.BooleanOptionalAction` once Python 3.9 is old enough.

Right... that too could work.

-----

In any case, the main benefit of `--unixsock` (noarg) is ~ease of typing on the CLI. If we make a sane default, and also make it configurable via config options (so that users don't need to type it for every command), then it does not matter if it is convenient to type.

- Let's just make the option a bit more generic now, before introducing it, so that it does not have to be changed later.
- Also, I like `rpcsock` more than `unixsock` as all(?) the exisiting local rpc options are named `rpc*` (`rpchost`, `rpcport`, `rpcuser`, `rpcpassword`). Relatedly, the `unixsock` name is somewhat ambiguous re what it refers to: is it the local rpc or is it the client-server communication?
- I guess you have a decent point that there might be no point in distinguishing between unix sockets and windows named pipes.

So, how about:

- `--rpcsock`, which takes one arg, valid arg values for now being `tcp` and `file`
    `parser.add_argument('rpcsock', choices=['tcp', 'file'])`
    later, if needed, we can split "file" up into separate options and default "file" to one of them
- `--rpcsockpath`, which also takes one arg, a file system path, used in case of `rpcsock==file`
- the respective config keys can be named identically to the cli/rpc options: `rpcsock` and `rpcsockpath`